packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -111,51 +92,25 @@ contract DowntimeSlasher is Ownable, Initializable, UsingRegistry, UsingPrecompi
     return true;
   }
 
+  /**
+   * @notice Returns the end block for the interval.
+   */
   function getEndBlock(uint256 startBlock) internal view returns (uint256) {
     return startBlock + slashableDowntime - 1;
   }
 
-  /**
-  function debugSlashed(address validator, uint256 startBlock)
-    public
-    view
-    returns (uint256, uint256, uint256, uint256)
-  {
-    uint256 endBlock = getEndBlock(startBlock);
-    uint256 startEpochBlock = (startBlock % getEpochSize());
-    uint256 endEpochBlock = endBlock % getEpochSize();
-    uint256 startEpoch = getEpoch(startBlock);
-    uint256 endEpoch = getEpoch(endBlock);
-    return (
-      startEpochBlock,
-      endEpochBlock,
-      isSlashed[keccak256(abi.encodePacked(validator, startEpoch))],
-      isSlashed[keccak256(abi.encodePacked(validator, endEpoch))]
-    );
-  }
-*/
-
   function checkIfAlreadySlashed(address validator, uint256 startBlock) internal {
     uint256 endBlock = getEndBlock(startBlock);
-    uint256 startEpochBlock = getEpochBlock(startBlock);
-    uint256 endEpochBlock = getEpochBlock(endBlock);
     uint256 startEpoch = getEpoch(startBlock);
     uint256 endEpoch = getEpoch(endBlock);
-    require(
-      isSlashed[keccak256(abi.encodePacked(validator, startEpoch))] < startEpochBlock,
-      "Already slashed"
-    );
-    require(
-      isSlashed[keccak256(abi.encodePacked(validator, endEpoch))] < endEpochBlock,
-      "Already slashed"
-    );
-    isSlashed[keccak256(abi.encodePacked(validator, startEpoch))] =
-      startEpochBlock +
-      slashableDowntime;
-    isSlashed[keccak256(abi.encodePacked(validator, endEpoch))] = endEpochBlock + 1;
+    require(isSlashed[validator][startEpoch] < startBlock, "Already slashed");
```

Comment:
What does two checks (and two sets) of `isSlashed` get us here, as opposed to just checking `isSlashed[validator][startEpoch] < startBlock` and setting `isSlashed[validator][startEpoch] = endBlock`?

Reply:
It's to handle the case where for example if epoch length is 100 and slashable downtime is 10, if there is a downtime starting at block 400 (epoch 4) and it lasts until block 410 (11 blocks downtime). In the current version that can only be slashed once, but in the simpler version it could be slashed twice. (But perhaps it would be better to slash twice so that both groups get slashed?)

Reply:
In this design it will depend on the order of calls anyway, so in your example, you'd call it 10 times with start blocks 50, 60, etc.
Changing the confusing name `isSlashed` to `lastSlashedBlock`.